homeassistant/core.py

```diff
@@ -847,8 +847,7 @@ def _cancel_cancellable_timers(self) -> None:
             if (
                 not handle.cancelled()
                 and (args := handle._args)  # pylint: disable=protected-access
-                # pylint: disable-next=unidiomatic-typecheck
-                and type(job := args[0]) is HassJob
+                and type(job := args[0]) is HassJob  # noqa: E721
```

Comment:
This is a sign of ruff's E721 likely not catching the comparison, added it just as a precaution to mark that ignoring it is intentional here. We don't currently flag unnecessary ruff configs as an error, so this should be fine.